core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -164,4 +164,7 @@ private TableProperties() {
 
   public static final String MERGE_MODE = "write.merge.mode";
   public static final String MERGE_MODE_DEFAULT = "copy-on-write";
+
+  public static final String MERGE_WRITE_CARDINALITY_CHECK = "write.merge.cardinality.check.enabled";
```

Comment:
I like Anton's original suggestion better than this property. How about `write.merge.cardinality-check.enabled`?